src/fs/file.rs

```diff
@@ -292,6 +292,59 @@ impl File {
+    /// necessary.
+    ///
+    /// Parameter `pos` is an `Option<u64>` to allow this function to be used for both files that
+    /// are seekable and those that are not. The caller is responsible for knowing this.
+    ///
+    /// When `None` is supplied, the offset passed to the `io_uring` call will always be zero, even
+    /// if multiple writev calls are necessary; only the iovec information would be adjusted
+    /// between calls. A Unix pipe would fall into this category.
+    ///
+    /// When `Some(n)` is suppied, the offset passed to the writev call will be incremented by the
+    /// progress of prior writev calls. A file system's regular file would fall into this category.
+    ///
+    /// If the caller passes `Some(n)` for a file that is not seekable, the `io_uring` `writev`
+    /// operation will return an error once n is not zero.
+    ///
+    /// If the caller passes `None`, when the file *is* seekable, when multiple `writev` calls are
+    /// required to complete the writing of all the bytes, the bytes at position 0 of the file will
+    /// have been overwritten one or more times with incorrect data. This is true just as if the
+    /// caller had invoked seperate write calls to a file, all with position 0, when in fact the
+    /// file was seekable.
+    ///
+    /// Performance considerations:
+    ///
+    /// The user may want to check that this function is necessary in their use case or performs
+    /// better than a series of write_all operations would. There is overhead either way and it is
+    /// not clear which should be faster or give better throughput.
+    ///
+    /// This function causes the temporary allocation of a Vec one time to hold the array of iovec
+    /// that is passed to the kernel. The same array is used for any subsequent calls to get all
+    /// the bytes written. Whereas individual calls to write_all do not require the Vec to be
+    /// allocated, they do each incur the normal overhead of setting up the submission and
+    /// completion structures and going through the future poll mechanism.
+    ///
+    /// TODO decide, would a separate `writev_all` function for `file` that did not take a `pos`
+    /// make things less ambiguous?
+    ///
+    /// TODO more complete documentation here.
+    /// TODO define writev_all functions for net/unix/stream, net/tcp/stream, io/socket.
+    /// TODO remove usize from result, to be consistent with other write_all_vectored functions.
+    /// TODO find a way to test this with some stress to the file so the writev calls don't all
+    /// succeed on their first try.
+    /// TODO consider replacing the current `write_all` and `write_all_at` functions with a similar
+    /// mechanism so all the write-all logic is in one place, in the io/write_all.rs file.
+    pub async fn writev_at_all<T: BoundedBuf>(
+        &self,
+        buf: Vec<T>,
+        pos: Option<u64>, // Use None for files that can't seek
```

Comment:
This looks like an API wart. I think most callers would prefer two different methods, the other being `writev_all` without an offset just like in other objects.